core/src/main/java/org/infinispan/commands/control/LockControlCommand.java

```diff
@@ -149,7 +148,7 @@ public void setParameters(int commandId, Object[] args) {
       globalTx = (GlobalTransaction) args[i++];
       unlock = (Boolean) args[i++];
       keys = (List<Object>) args[i++];
-      flags = (Set<Flag>) args[i];
+      flags = (EnumSet<Flag>) args[i];
```

Comment:
This change seems unneeded, did you want to change the declared type of flags as well?  Otherwise I could see someone possibly passing in the wrong set impl.